packages/wallets/nightly/src/adapter.ts

```diff
@@ -0,0 +1,150 @@
+
+    private _connecting: boolean;
+    private _connected: boolean;
+    private _publicKey: PublicKey | null;
+    private _wallet: SolanaNightly | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor() {
+        super();
+        this._connecting = false;
+        this._connected = false;
+        this._publicKey = null;
+        this._wallet = null;
+
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            scopePollingDetectionStrategy(() => {
+                if (window?.nightly?.solana) {
+                    this._readyState = WalletReadyState.Installed;
+                    this.emit('readyStateChange', this._readyState);
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+
+    get connected(): boolean {
+        return this._connected;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    public async signAllTransactions(transactions: Transaction[]): Promise<Transaction[]> {
```

Comment:
nit: could you reorder the methods to match the other adapters? It just makes review and maintenance of all these easier.

Reply:
done